torch_geometric/nn/conv/sage_conv.py

```diff
@@ -92,7 +119,19 @@ def forward(self, x: Union[Tensor, OptPairTensor], edge_index: Adj,
     def message(self, x_j: Tensor) -> Tensor:
         return x_j
 
-    def message_and_aggregate(self, adj_t: SparseTensor,
-                              x: OptPairTensor) -> Tensor:
+    def message_and_aggregate(self, adj_t: SparseTensor, x: OptPairTensor,
+                              edge_index_j, edge_index_i) -> Tensor:
+        """
+            Performs both message passing and aggregation of messages from neighbours using the aggregator_type
+        """
         adj_t = adj_t.set_value(None, layout=None)
-        return matmul(adj_t, x[0], reduce=self.aggr)
+        if self.aggregator_type == 'mean' or self.aggregator_type == 'gcn':
+            return matmul(adj_t, x[0], reduce='mean')
+
+        elif self.aggregator_type == 'max':
+            return matmul(adj_t, x[0], reduce='max')
+
+        elif self.aggregator_type == 'lstm':
+            x_j = x[0][edge_index_j]
+            x, mask = to_dense_batch(x_j, edge_index_i)
+            return self.lstm(x)
```

Comment:
Doesn't `self.lstm` return a tuple?

Reply:
Yes, correct. I just tested it out and it is returning a tuple and throwing this error: 
`linear(): argument 'input' (position 1) must be Tensor, not tuple`

Earlier when I was testing out, I realised that I did not transform the dataset with ToSparseTensor() and hence it was going to the regular message(), aggregate(), update() flow instead of going to message_and_aggregate().